api/serializers.py

```diff
@@ -0,0 +1,276 @@
+    class Meta:
+        model = Partnership
+        fields = [
+            'url', 'partner', 'supervisor', 'ucl_university', 'ucl_university_labo',
+            'ucl_sector', 'is_sms', 'is_smp', 'is_smst', 'is_sta', 'is_stt',
+            'education_fields', 'status', 'partner_entity', 'medias',
+            # OUT
+            'out_education_levels', 'out_entities', 'out_university_offers',
+            'out_contact', 'out_portal', 'out_funding',
+            'out_partner_contacts',
+            # IN
+            'in_contact', 'in_portal',
+            # STAFF
+            'staff_contact_name', 'staff_funding',
+            'staff_partner_contacts',
+        ]
+
+    def _get_current_year_attr(self, partnership, attr):
+        try:
+            return getattr(partnership.current_year_for_api[0], attr)
+        except IndexError:
+            return None
+
+    def get_is_sms(self, partnership):
+        return self._get_current_year_attr(partnership, 'is_sms')
+
+    def get_is_smp(self, partnership):
+        return self._get_current_year_attr(partnership, 'is_smp')
+
+    def get_is_smst(self, partnership):
+        return self._get_current_year_attr(partnership, 'is_smst')
+
+    def get_is_sta(self, partnership):
+        return self._get_current_year_attr(partnership, 'is_sta')
+
+    def get_is_stt(self, partnership):
+        return self._get_current_year_attr(partnership, 'is_stt')
+
+    def get_education_fields(self, partnership):
+        education_fields = self._get_current_year_attr(partnership, 'education_fields')
+        if education_fields is None:
+            return None
+        return ['{0} ({1})'.format(field.label, field.code) for field in education_fields.all()]
+
+    def get_status(self, partnership):
+        if partnership.agreement_status is None:
+            return None
+        return {
+            'status': partnership.agreement_status,  # annotation on the queryset
+            'valid_years': partnership.validity_years,  # annotation on the queryset
+        }
+
+    def get_out_education_levels(self, partnership):
+        education_levels = self._get_current_year_attr(partnership, 'education_levels')
+        if education_levels is None:
+            return None
+        return [level.label for level in education_levels.all()]
+
+    def get_out_entities(self, partnership):
+        entities = self._get_current_year_attr(partnership, 'entities')
+        if entities is None:
+            return None
+        return [str(entity) for entity in entities.all()]
+
+    def get_out_university_offers(self, partnership):
+        offers = self._get_current_year_attr(partnership, 'offers')
+        if offers is None:
+            return None
+        return ['{} - {}'.format(offer.acronym, offer.title) for offer in offers.all()]
+
+    def get_out_contact(self, partnership):
+        contact = {
+            'email': getattr(partnership.ucl_management_entity, 'contact_out_email', None),
+            'title': None,
```

Comment:
Le "title" est-il pertinent? (Il est toujours initialisé à None)

Reply:
Ça correspond au schema : les contacts `in` et `out` sont des `Person` dans le back mais en front/schema on a qu'une entité `Contact` pour simplifier la structure